src/p2p/net_node.inl

```diff
@@ -1776,19 +1810,36 @@ namespace nodetool
   bool node_server<t_payload_net_handler>::set_max_out_peers(const boost::program_options::variables_map& vm, int64_t max)
   {
     if(max == -1) {
-      m_config.m_net_config.connections_count = P2P_DEFAULT_CONNECTIONS_COUNT;
+      m_config.m_net_config.max_out_connection_count = P2P_DEFAULT_CONNECTIONS_COUNT;
+      return true;
+    }
+    m_config.m_net_config.max_out_connection_count = max;
+    return true;
+  }
+
+  template<class t_payload_net_handler>
+  bool node_server<t_payload_net_handler>::set_max_in_peers(const boost::program_options::variables_map& vm, int64_t max)
+  {
+    if(max == -1) {
+      m_config.m_net_config.max_in_connection_count = P2P_DEFAULT_CONNECTIONS_COUNT;
```

Comment:
That change causes a large change in the network, no ?
I think that if a new node comes online, it might not be able to connect to the network (or will take a while to find a node which doesn't have all its slots free). I would either consider the -1 default to mean no limit, or set a substantially larger limit than the out one.

Reply:
Set to `-1`.